test/lib/generate-mock-query.js

```diff
@@ -0,0 +1,49 @@
+const {
+  lib: {
+    index: { isString },
+  },
+} = require('@0ti.me/tiny-pfp');
+const {
+  _,
+  sinon: { spy },
+} = deps;
+
+module.exports = function() {
+  let responses = [];
+
+  this.bindQueryAndResponse = (query, response) => {
+    responses.push({ query, response });
+
+    return this;
+  };
+
+  this.query = spy(query => {
+    const result = responses.reduce((acc, each) => {
+      if (acc) return acc;
+
+      if (isString(each.query)) {
+        if (query === each.query) {
+          return each.response;
+        }
+      } else if (each.query.test) {
+        if (each.query.test(query)) {
```

Comment:
this one's a little weird - if `each.query` has a function `test` that returns true, and if running that method with the target query passed in as a parameter returns true, then return the mocked response for the query we're looking for?

Reply:
Ah, this is really bad code, especially in a test library where I don't care if it's bulky. I should be using:
```
} else if (lodash.isRegExp(each.query)) {
  if (each.query.test(query)) { // if a match is found on the string (`query`) which was actually used to call the `mockQuery` function and given the regexp (`each.query`)
    return response; // return the matching response
```

I don't handle the fallthrough because that indicates the regexp didn't find a match and I want to keep searching for the matching `each.query` regular expression or string if one exists.